controllers/core/openstackcontrolplane_controller.go

```diff
@@ -110,6 +112,25 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, err
 	}
 
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
```

Comment:
nit: logging something for a reconcile failure before returning nil is a good practice

Reply:
The error is already logged because it is returned, but we can add additional context around it